Skip fault and debug logs when logd is blocked

pvollan opened the pull request:
#### 3022cdc9fa3b578bf8c4842b7989e9dd72b6dc28
<pre>
Skip fault and debug logs when logd is blocked
<a href="https://bugs.webkit.org/show_bug.cgi?id=257211">https://bugs.webkit.org/show_bug.cgi?id=257211</a>
rdar://109723241

Reviewed by Brent Fulgham.

Skip fault and debug logs when logd is blocked in the sandbox, since logd is currently only blocked in
non internal builds.

* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.cpp:
(WebKit::NetworkConnectionToWebProcess::logOnBehalfOfWebContent):
* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
(WebKit::registerLogHook):

Canonical link: <a href="https://commits.webkit.org/264584@main">https://commits.webkit.org/264584@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/00e74e1b2d1d701de9d38bb6d462dcfe7f70debe

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/7915 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/8193 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/8442 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/9564 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/8038 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/10186 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/8110 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/10904 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/8055 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/9155 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/7279 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/9682 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/6459 "Passed tests") | | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/14843 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/7582 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/7403 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/10737 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/7834 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/6369 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/7152 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1928 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/11361 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/7569 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
```diff
@@ -676,35 +676,24 @@ static void registerLogHook()
         if (msg->buffer_sz > 1024)
             return;
 
+        // Skip faults and debug logs in non-internal builds.
```
Should this be enclosed in `#if USE(APPLE_INTERNAL_SDK)` or something?

pvollan replied:
That is a good point. In this case I do not think it is needed, since this code will only run on non internal builds.

brentfulgham commented on Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
```diff
@@ -676,35 +676,24 @@ static void registerLogHook()
-            if (!messageString)
-                return;
-            IPC::DataReference logString(reinterpret_cast<uint8_t*>(messageString), strlen(messageString) + 1);
+        char* messageString = os_log_copy_message_string(&msg);
```
auto messageString = std::unique_ptr<char*, decltype(std::free) *>(os_log_copy_message_string(&msg), std::free);

brentfulgham commented on Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
```diff
@@ -676,35 +676,24 @@ static void registerLogHook()
 
-            free(messageString);
-        });
+        free(messageString);
```
If you do the `std::unique_ptr` thing above, you don't need to have the explicit `free` here.

pvollan replied:
> If you do the `std::unique_ptr` thing above, you don't need to have the explicit `free` here.

Fixed!

Thanks for reviewing!

brentfulgham reviewed (approved):
Looks good to me.

pvollan committed:
Skip fault and debug logs when logd is blocked
https://bugs.webkit.org/show_bug.cgi?id=257211
rdar://109723241

Reviewed by Brent Fulgham.

Skip fault and debug logs when logd is blocked in the sandbox, since logd is currently only blocked in
non internal builds.

* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.cpp:
(WebKit::NetworkConnectionToWebProcess::logOnBehalfOfWebContent):
* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
(WebKit::registerLogHook):

Canonical link: https://commits.webkit.org/264584@main
